src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -407,29 +410,33 @@ void Chttp2ServerListener::ActiveConnection::HandshakingState::Start(
                              OnHandshakeDone, this);
 }
 
-void Chttp2ServerListener::ActiveConnection::HandshakingState::OnTimeout(
-    void* arg, grpc_error_handle error) {
-  HandshakingState* self = static_cast<HandshakingState*>(arg);
-  // Note that we may be called with absl::OkStatus() when the timer fires
-  // or with an error indicating that the timer system is being shut down.
-  if (error != absl::CancelledError()) {
+void Chttp2ServerListener::ActiveConnection::HandshakingState::OnTimeout() {
+  grpc_chttp2_transport* transport = nullptr;
+  {
+    MutexLock lock(&connection_->mu_);
+    if (timer_handle_.has_value()) {
+      transport = connection_->transport_;
+      timer_handle_.reset();
+    }
+  }
+  if (transport) {
```

Comment:
Please write `!= nullptr`, to make it clear to the reader that this is a pointer, not a boolean.